### PR TITLE
New Emag Interaction for Cyborg Rechargers

### PR DESCRIPTION
**Pull request by Runian**
# Document the changes in your pull request
The cyborg recharger now has a new interaction with an emag. It can be used to auto-borg people inside of it. Only usable by people with arms (non-silicons).

Victim (aka the occupant of the recharger) must satisfy the following conditions to become auto-borged:
- Cannot be a cyborg.
- Cannot have pierce immunity.
- Cannot be a changeling.
- Must be not dead.

There is a 30-second period in which the process is "loading" in which the victim can attempt to escape. It takes 4 seconds to do so. 
If the victim does not escape, the process begins:
- Paralyzed for the rest of the process.
- Failing the auto-borg conditions stops the borging mid-process.
- All removable non-robotic arms and legs are dismembered. Each limb delays the borging by 2 seconds and makes alot of gibs.
- They become a cyborg with a 5000 power cell and automatically links up to an AI if there is one.
- Stunned for 4 seconds after becoming a cyborg.

It takes 2 seconds to activate the auto-borg feature.

The final process looks like this (with old sprite):
autoborgin

# Spriting
When in use: 
newsprite

# Wiki Documentation
Add this to the wiki's [Emag](https://wiki.yogstation.net/wiki/Emag) page. Recommend usage of this on people who are either asleep, soft-crit organics, or hard-crit non-organics. Usage not recommended on hard-crit organics since dismemberments can kill.

# Changelog

:cl:  
rscadd: Cyborg recharger now has emag functionality which causes it becomes an ineffective auto-borger.
/:cl:

### CODE REVIEW

**Comment by GenericDM** [comment]
I'd suggest a special sprite state, potentially similar to hacked suit storage units.

File: code/game/machinery/rechargestation.dm [hunk 1]
```diff
@@ -28,26 +38,158 @@
 	for(var/obj/item/stock_parts/cell/C in component_parts)
 		recharge_speed *= C.maxcharge / 10000
 
+/obj/machinery/recharge_station/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return
+	obj_flags |= EMAGGED
+	playsound(src, "sparks", 75, TRUE, -1)
+	to_chat(user, span_notice("You use the cryptographic sequencer on [src] causing a well-hidden panel to open."))
```
Comment (GenericDM) [inline code/game/machinery/rechargestation.dm]: This implies that NT standard chargers are just meant to do this. Maybe reflavor to a hijacking of the borg repair gear inside?

Comment (Runian) [reply code/game/machinery/rechargestation.dm]: done

**Comment by RG4ORDR** [comment]
No.

**Comment by Runian** [comment]
> No.

If I made it a roboticist-exclusive item in the uplink, 12/16 TC, and/or hijack only, would you change your mind?

**Comment by RG4ORDR** [comment]
> > No.
> 
> If I made it a roboticist-exclusive item in the uplink, 12/16 TC, and/or hijack only, would you change your mind?

I just can't.

This item has one those auras of - totally shit but one round in the future one person will pull it off and never see use like that again. Traitor robo has access to some extremely strong traitor items already and fast and smart traitor robos can assemble combat mechs and have a double buff provided by a suit + book and only really need to spend 11 TCs vs this.

**Comment by Runian** [comment]
> > > No.
> > 
> > 
> > If I made it a roboticist-exclusive item in the uplink, 12/16 TC, and/or hijack only, would you change your mind?
> 
> I just can't.
> 
> This item has one those auras of - totally shit but one round in the future one person will pull it off and never see use like that again. Traitor robo has access to some extremely strong traitor items already and fast and smart traitor robos can assemble combat mechs and have a double buff provided by a suit + book and only really need to spend 11 TCs vs this.

There are already enough busted traitors items that are bad when used poorly, but can be amazingly good:
-  Hypnotic Flash costs 7 TCs, on-par with a mindslave and slightly more expensive than a brainwashing disk. It gives a detectable brain trauma that can be only cured via surgery (which they'll resist the entire way there). Can be easy to use since all it takes is 1u of mindbreaker or being asleep. 
- Dual wielding revolvers and two speedloaders for 13 TCs. Amazing burst damage and has near-infinite ammo from a hacked autolathe that allows for near-endless shooting and ignoring airlocks since you can dump your revolvers into the airlock until it breaks.
- X-ray bow and thermals for 10 TCs. Infinite ammo bow that has good damage, fast reload speed for a bow, gives enough rads to cause toxin damage after 2 hits, goes through walls, has a scope feature so you can snipe from further away, and lets you do all of this in relative safety with thermals.
- Full damage ranged Holoparasite for 15 TCs. Infinite ammo, shoots at a decent speed, decent damage, and can punch open most things open (like instantly breaking walls).
- Powerfist with 2533kpa tank for 6 TCs. Crits people in 2 hits, even people with deathsquad levels of armor.
- 100u meth explosion explosive spear for 2 TCs (makeshift crafting). Gibs the person who is hit with it.
- Airshoes for 4 TCs. Incredibly fast speed and it is entirely possible to negate the bump-stun that is suppose to punish for not being careful.

Anyways, it would be nice to have something that a traitor roboticist can do that isn't the usual "must build very strong durand and kill everyone who dares look at me wrong" like the other aspects of their job which are robots/cyborgs. Again, willing to up the price or make it an item that isn't the emag if it means for a fairer price.

**Comment by ynot01** [comment]
maybe if it was own traitor item but i doubt it, just seems like a bad idea

**Pull request closed without merging**